src/app/shared/components/guided-tour/dialog/template.html

```diff
@@ -0,0 +1,17 @@
+<!--
+Copyright 2020 The Kubermatic Kubernetes Platform contributors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<div class="km-guided-tour">
+  <km-gt-add-project-btn *ngIf="step.id === GuidedTourID.AddProjectBtn"></km-gt-add-project-btn>

```

Comment:
My initial idea when talking about dynamic steps was to use `ComponentFactory`. Registering a new step would be then as simple as adding a single entry to the `id -> component class` registry.